packages/chevrotain/src/parse/parser/traits/gast_recorder.ts

```diff
@@ -412,7 +412,7 @@ function recordOrProd(mainProdArg: any, occurrence: number): any {
   forEach(alts, (currAlt) => {
     const currAltFlat = new Alternative({ definition: [] })
     newOrProd.definition.push(currAltFlat)
-    if (has(currAlt, "IGNORE_AMBIGUITIES")) {
+    if (currAlt.IGNORE_AMBIGUITIES !== undefined) {
```

Comment:
was this change needed to enable strictNullChecks?

Reply:
For the case `{IGNORE_AMBIGUITIES: myBoolean}` and `{}` this is semantically identical. For the case `{IGNORE_AMBIGUITIES: undefined}`, the existing code would then assign undefined to `currAltFlat.ignoreAmbiguities`, which is wrong, as that is defined as a boolean.

I did not change occurrences of `has` which did not result in an error in an attempt to minimize code changes.

Supressing the error with `currAltFlat.ignoreAmbiguities = currAlt.IGNORE_AMBIGUITIES!` would be possible but incorrect as it can actually be undefined. As these are the kind of errors strictNullChecks is designed to avoid, fixing it is much preferable to keeping a consistent use of has.

Reply:
> Supressing the error with currAltFlat.ignoreAmbiguities = currAlt.IGNORE_AMBIGUITIES! would be possible but incorrect as it can actually be undefined. As these are the kind of errors strictNullChecks is designed to avoid, fixing it is much preferable to keeping a consistent use of has.

As mentioned earlier, these are user inputted values, so the error can always happen (e.g not all consumers of the library use typescript). So I'd rather avoid semantic changes in this PR. (even tiny ones).

Reply:
i fixed this